### PR TITLE
feat: nuget docs

### PR DESCRIPTION
**Pull request by aryehbeitz**
Docs for NuGet package manager.
I used `.NET` instead of `NuGet` when I saw languages (instead of package managers), such as `Go` specified

### CODE REVIEW

File: _docs_nuget/1_support.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+title: "What type of NuGet projects does Snyk support?"
+---
+Snyk supports testing and monitoring GNuGet projects that have their dependencies managed by [nuget](https://nuget.org). NuGet support is currently available via the Snyk [CLI](/docs/using-snyk/).
```
Comment (michael-go) [inline _docs_nuget/1_support.md]: Mention also that it's still in Beta

**Comment by michael-go** [comment]
@aryehbeitz  note you'll also to make some additions in `registry` for all the docs to show, like in this PR for Go: https://github.com/snyk/registry/pull/3458

File: _docs_nuget/1_support.md (hunk 1 above)
Comment (anermazur) [reply _docs_nuget/1_support.md]: GNuGet projects? typo or as should be?

**Comment by aryehbeitz** [comment]
@michael-go added
https://github.com/snyk/registry/pull/3485

File: _docs_nuget/2_testing.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+title: "Testing .NET projects"
+---
+
+We scan .NET projects managed by NuGet, by examining your `project.json`, `obj/project.assets.json`, `packages.config` and `.csproj` files to compare the specific versions of every [direct and deep dependency](https://snyk.io/docs/faqs/#about-known-vulnerabilities) in your project against our [NuGet vulnerability database](/vuln?type=nuget).
```
Comment (adrukh) [inline _docs_nuget/2_testing.md]: Remove reference to `project.json` and `packages.config`.
Change `.csproj` to `*.csproj`.
Mention .NET Core being related to `obj/project.assets.json`, `*.csproj` related to regular .NET projects. Is there a common name for non-Core .NET?

File: _docs_using-snyk/03_test.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ title: Test
 <p><code>snyk test</code> takes stock of all the local dependencies and queries the snyk service for related known vulnerabilities. It displays the found issues along with additional information. For Node.js projects, it also suggests remediation steps.</p>
 
 <p>When <code>snyk test</code> runs, it tries to detect the appropriate file for your project by looking for the following files, in this order:</p>
-<ol><li>yarn.lock</li><li>package.json</li><li>Gemfile</li><li>Gemfile.lock</li><li>pom.xml</li><li>requirements.txt</li><li>build.gradle</li><li>build.sbt</li><li>Gopkg.lock</li><li>vendor/vendor.json</li></ol>
+<ol><li>yarn.lock</li><li>package.json</li><li>Gemfile</li><li>Gemfile.lock</li><li>pom.xml</li><li>requirements.txt</li><li>build.gradle</li><li>build.sbt</li><li>Gopkg.lock</li><li>vendor/vendor.json</li><li>project.json</li><li>project.assets.json</li><li>packages.config</li><li>.csproj</li></ol>
```
Comment (adrukh) [inline _docs_using-snyk/03_test.md]: Update this to include only `obj/project.assets.json` and `*.csproj` - this is autodetection as we discussed.

File: _docs_using-snyk/03_test.md [hunk 4]
```diff
@@ -25,6 +25,12 @@ Since <code>snyk test</code> looks at the locally installed modules, it needs to
 
 **Note for Scala:** <br> In order to use the CLI to test against your `build.sbt` manifest file, you'll need to first install the [sbt-dependency-graph plugin](https://github.com/jrudolph/sbt-dependency-graph).
 
+<p><strong>Note for Golang:</strong> <br>
+Since <code>snyk test</code> looks at the locally installed modules, it needs to run after the <code>vendor/</code> was populated via <code>dep ensure</code> or <code>govendor sync</code>. In addition, the <code>GOPATH</code> environment variable must be set correctly.</p>
```
Comment (adrukh) [inline _docs_using-snyk/03_test.md]: Doesn't this get added with the Golang docs?

File: _docs_using-snyk/08_dev-integration.md [hunk 5]
```diff
@@ -11,7 +11,7 @@ title: Integrating Snyk into your dev workflow
 3. Ensure the `.snyk` file you generated was added to your source control (`git add .snyk`).
 4. If you selected to, Snyk will include `snyk test` as part of your `npm test` command, so if there are new vulnerabilities in the future, your CI will fail, protecting you from introducing vulnerabilities to production. Alternatively, you can add `snyk test` to any other CI test platform you use.
 
-<h3>For Ruby, Python, Java and Go CI</h3>
+<h3>For Ruby, Python, Java, Go CI and .NET</h3>
```
Comment (adrukh) [inline _docs_using-snyk/08_dev-integration.md]: `... Go and .NET CI`

File: _docs_nuget/2_testing.md (hunk 2 above)
Comment (anermazur) [reply _docs_nuget/2_testing.md]: Could we add here also the points mentioned before, on what needs to be executed before running on each of the envs?

File: _docs_using-snyk/03_test.md [hunk 6]
```diff
@@ -25,6 +25,12 @@ Since <code>snyk test</code> looks at the locally installed modules, it needs to
...
+
+<p><strong>Note for .NET:</strong> <br>
+Since <code>snyk test</code> looks at the locally installed modules, it needs to run after the <code>packages/</code>(.NET) or <code>obj/</code>(.NET Core) has been populated via Visual Studio or <code>dotnet restore</code>.</p>
+
 Running `snyk test` on your Scala projects without this plugin will throw the following error:
```
Comment (adrukh) [inline _docs_using-snyk/03_test.md]: This belongs to the Scala part, make sure you don't break it.

File: _docs_using-snyk/03_test.md (hunk 4 above)
Comment (aryehbeitz) [reply _docs_using-snyk/03_test.md]: my mistake. removed

File: _docs_nuget/1_support.md [hunk 7]
```diff
@@ -0,0 +1,4 @@
+---
+title: "What type of .NET projects does Snyk support?"
+---
+Snyk supports testing and monitoring .NET projects that have their dependencies managed by [NuGet](https://nuget.org). .NET support, (currently in Beta stage), is available via the Snyk [CLI](/docs/using-snyk/).
```
Comment (adrukh) [inline _docs_nuget/1_support.md]: The file needs to be named `_docs_dotnet/1_support.md`, please rename all others.

**Pull request merged**